e2e/browser/src/components/notifications/index.tsx

```diff
@@ -21,13 +21,127 @@ const MessageList = (props: { messages: Array<any> }) => {
   );
 };
 
+const WebSocketButtons = ({ socket }: { socket?: WebsocketNotification }) => {
+  if (socket === undefined) {
+    return <></>;
+  }
+  return (
+    <>
+      <button
+        onClick={async (e) => {
+          e.preventDefault();
+          await socket.connect();
+        }}
+        data-testid="connectSocket"
+      >
+        Connect websocket
+      </button>
+      <button
+        onClick={(e) => {
+          e.preventDefault();
+          socket.disconnect();
+        }}
+        data-testid="disconnectSocket"
+      >
+        Disconnect websocket
+      </button>
+    </>
+  );
+};
+
+const CreateResourceButton = ({
+  parentContainerUrl,
+  setChildContainerUrl,
+}: {
+  parentContainerUrl?: string;
+  setChildContainerUrl: (url: string) => void;
+}) => {
+  return parentContainerUrl === undefined ? (
+    <></>
+  ) : (
+    <button
+      onClick={async (e) => {
+        e.preventDefault();
+        setChildContainerUrl(
```

Comment:
94c88acb7165ce48755c1b4a76676aea304a1cf1

Reply:
Do you have resources in mind explaining why this is an anti-pattern, or alternatively would you mind having a chat sometime to walk me through the issue so that I understand it better ?